core/math/a_star_grid_2d.cpp

```diff
@@ -377,89 +361,101 @@ void AStarGrid2D::_get_nbors(Point *p_point, List<Point *> &r_nbors) {
 	}
 }
 
-bool AStarGrid2D::_solve(Point *p_begin_point, Point *p_end_point) {
-	pass++;
-
+bool AStarGrid2D::_solve(const Point *p_begin_point, const Point *p_end_point, List<const Point *> &r_path) const {
 	if (p_end_point->solid) {
 		return false;
 	}
 
-	bool found_route = false;
-
-	Vector<Point *> open_list;
-	SortArray<Point *, SortPoints> sorter;
+	LocalVector<Pass *> open_list;
+	SortArray<Pass *, SortPasses> sorter;
 
-	p_begin_point->g_score = 0;
-	p_begin_point->f_score = _estimate_cost(p_begin_point->id, p_end_point->id);
-	open_list.push_back(p_begin_point);
-	end = p_end_point;
+	RBMap<Vector2i, Pass> data;
+	open_list.push_back(&data.insert(p_begin_point->id, Pass(p_begin_point, 0, _estimate_cost(p_begin_point->id, p_end_point->id)))->get());
```

Comment:
Note: using `RBMap` instead of `HashMap` increased the performance for about 0.0001ms, using stack-allocated Pass instead of heap increase the performance from 0.0025 to 0.0012. These tests are made by using 1000x1000 grid to find path from 0,0 to 999,999 without jumping to calculate average time using 1000 attempts, using the following code:

```
	var astar_grid = AStarGrid2D.new()
	astar_grid.size = Vector2i(1000, 1000)
	astar_grid.cell_size = Vector2(16, 16)
	astar_grid.update()

	var average = 0.0
	for t in range(1000):
		var ms = Time.get_unix_time_from_system()
		var path = astar_grid.get_id_path(Vector2i(0, 0), Vector2i(999, 999))
		average += Time.get_unix_time_from_system() - ms
	print(average / 1000.0)
```